### PR TITLE
clarify a comment on annotation key validation

### LINKED ISSUE #109459: The validation of Annotations
https://github.com/kubernetes/kubernetes/blob/c285e781331a3785a7f436042c65c5641ce8a9e9/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L50

Why do we need `strings.ToLower(k)`?

### PR DESCRIPTION
**Pull request by jlsong01**
#### What type of PR is this?
/kind cleanup
/sig api-machinery

#### What this PR does / why we need it:
Clarify a comment on annotation key validation. 
Since uppercase is valid for annotation key but not QualifiedName, it's better to clarify it for avoid confusion and code readability.

#### Which issue(s) this PR fixes:

Fixes #109459

#### Special notes for your reviewer:
None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ var BannedOwners = map[schema.GroupVersionKind]struct{}{
 func ValidateAnnotations(annotations map[string]string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for k := range annotations {
+		// Uppercase is valid for annotation key but not QualifiedName, convert it as lowercase
```
Comment (roycaihw) [inline staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go:47]: nit: IIUC, in particular, uppercase is invalid in a QualifiedName's **prefix**:
- `a.b/C` is a valid QualifiedName
- `A.b/C` is not

The metadata.annotation [doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set) says "the prefix must be a DNS subdomain" which is a little unclear. DNS is case insensitive and can be normalized. While a QualifiedName is expected to have a [lowercase](https://github.com/kubernetes/kubernetes/blob/cae22d8b8a78985f438c232357aa2b6c60d83f9b/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L200) (normalized) RFC 1123 subdomain prefix, today we allow uppercase in metadata.annotation's prefix.

**Comment by jlsong01** [comment]
/assign @lavalamp 
for approval

File: staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go (hunk 1 above)
Comment (lavalamp) [inline staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go:47]:
```suggestion
		// The rule is QualifiedName except that case doesn't matter, so convert to lowercase before checking.
```

Comment (jlsong01) [reply staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go]: Thanks for the suggestioin. Fixed.

**Pull request merged**